src/gui/guiEngine.cpp

```diff
@@ -612,3 +612,40 @@ unsigned int GUIEngine::queueAsync(const std::string &serialized_func,
 	return m_script->queueAsync(serialized_func, serialized_params);
 }
 
+/******************************************************************************/
+bool GUIEngine::CheckAbsPath(const std::string &path, bool *write_allowed)
+{
+	if (path.empty())
+		return false;
+
+	if (fs::PathStartsWith(path, fs::TempPath()) ||
+			fs::PathStartsWith(path, fs::RemoveRelativePathComponents(porting::path_user + DIR_DELIM "games")) ||
+			fs::PathStartsWith(path, fs::RemoveRelativePathComponents(porting::path_user + DIR_DELIM "mods")) ||
+			fs::PathStartsWith(path, fs::RemoveRelativePathComponents(porting::path_user + DIR_DELIM "textures")) ||
+			fs::PathStartsWith(path, fs::RemoveRelativePathComponents(porting::path_user + DIR_DELIM "worlds")) ||
+			fs::PathStartsWith(path, fs::RemoveRelativePathComponents(porting::path_cache))) {
+		if (write_allowed) *write_allowed = true;
```

Comment:
```suggestion
		if (write_allowed)
			*write_allowed = true;
```
https://dev.minetest.net/Code_style_guidelines#If_statements
>Violating this rule will result in INSTANT REJECTION! **STOP DOING IT!**

Reply:
how bout this
```C++
(write_allowed && *write_allowed = true);
```

Reply:
> how bout this
> 
> ```c++
> (write_allowed && *write_allowed = true);
> ```

Huh, is this even possible in c++?

`write_allowed ? *write_allowed = true :;` might work.

Reply:
Oh sorry. Forgot the parentheses.  https://www.tutorialspoint.com/compile_cpp_online.php
```C++
    bool test = false;
    bool *test_ptr = &test;
    (test_ptr && (*test_ptr = true));

    cout << (test ? "yes" : "no") << endl; 
```